### DISCUSSION TITLE
Pushing notifications to android

### DISCUSSION THREAD

**Discussion by Yogesh-Dubey-Ayesavi** [processone/ejabberd#4061] [Questions] [answered]
Well, I am creating an android chatting application, I want to push notifications to those users who are currently to active, through google's [fcm](https://firebase.google.com/docs/cloud-messaging)  how can I implement that?  if it isn't possible, are there any other mechanisms to perform such operations?

Well I checked some issues earlier, i found
 ```
mod_push :{}

 mod_push_keepalive: {}
```
But i dont know how to use them can someone give some pointers.
Thanks in advance.

**Comment by licaon-kter** [comment]
You have a special app that you published on Play or App Store, or you use Conversations or Monal or some app that are on Play or App Store?

**Comment by Yogesh-Dubey-Ayesavi** [comment]
> You have a special app that you published on Play or App Store, or you use Converasations or Monal or some app that are on Play or App Store?

I haven't published it yet just playing around with it and testing in local environments. 

What I want to implement is, whenever the user stops listening to the xmpp server for instance terminates the application, and in that time if some other user messages him, I want to push a notification to that user who stopped listening to the xmpp server regarding the new message mostly what all chatting applications do. 

Well I'm having my own servers to push notifications what I want is how to trigger them if a user goes offline and receives a message. 

Thanks and regards.

**Comment by licaon-kter** [comment, marked as answer]
Great, so you need an XMPP server like ejabberd, and a component to talk with Google/Apple, like https://github.com/iNPUTmice/p2 or https://github.com/monal-im/fpush 

The message sending happens by default, but... depending how the app "stopped" say by user putting it in background, OS suspending, user killing the app, OS killing the app, they might get the message via push... or not.
